multiwallet.go

```diff
@@ -56,80 +52,62 @@ func NewMultiWallet(rootDir, dbDriver, netType, politeiaHost string) (*MultiWall
 		return nil, err
 	}
 
-	rootDir = filepath.Join(rootDir, netType)
-	err = os.MkdirAll(rootDir, os.ModePerm)
-	if err != nil {
-		return nil, errors.Errorf("failed to create rootDir: %v", err)
-	}
-
-	err = initLogRotator(filepath.Join(rootDir, logFileName))
-	if err != nil {
-		return nil, errors.Errorf("failed to init logRotator: %v", err.Error())
-	}
-
-	mwDB, err := storm.Open(filepath.Join(rootDir, walletsDbName))
-	if err != nil {
-		log.Errorf("Error opening wallets database: %s", err.Error())
-		if err == bolt.ErrTimeout {
-			// timeout error occurs if storm fails to acquire a lock on the database file
-			return nil, errors.E(ErrWalletDatabaseInUse)
-		}
-		return nil, errors.Errorf("error opening wallets database: %s", err.Error())
-	}
-
-	// init database for saving/reading wallet objects
-	err = mwDB.Init(&Wallet{})
-	if err != nil {
-		log.Errorf("Error initializing wallets database: %s", err.Error())
-		return nil, err
-	}
-
-	// init database for saving/reading proposal objects
-	err = mwDB.Init(&Proposal{})
+	dcrDB, dcrRootDir, err := initializeDCRWallet(rootDir, dbDriver, netType)
 	if err != nil {
-		log.Errorf("Error initializing wallets database: %s", err.Error())
-		return nil, err
+		log.Errorf("error initializing DCRWallet: %s", err.Error())
+		return nil, errors.Errorf("error initializing DCRWallet: %s", err.Error())
 	}
 
 	mw := &MultiWallet{
-		dbDriver:    dbDriver,
-		rootDir:     rootDir,
-		db:          mwDB,
-		chainParams: chainParams,
-		wallets:     make(map[int]*Wallet),
-		badWallets:  make(map[int]*Wallet),
-		syncData: &syncData{
-			syncProgressListeners: make(map[string]SyncProgressListener),
+		DbDriver:    dbDriver,
+		RootDir:     dcrRootDir,
+		DB:          dcrDB,
+		ChainParams: chainParams,
+		Assets: &Assets{
+			DCR: struct {
+				Wallets     map[int]*dcr.Wallet
+				BadWallets  map[int]*dcr.Wallet
+				DBDriver    string
+				RootDir     string
+				DB          *storm.DB
+				ChainParams *chaincfg.Params
```

Comment:
These aren't set or used yet. Can be removed till if and when they're needed.

Reply:
![Screenshot from 2022-08-26 15-47-46](https://user-images.githubusercontent.com/25265396/186931624-d9111a85-2e33-4592-bd38-aa773a9e5df7.png)

Reply:
db driver is not used, would remove that, the others are being used